updating-bazel.md

```diff
@@ -0,0 +1,57 @@
+---
+layout: contribute
+title: Updating Bazel
+---
+
+# Updating Bazel
+
+Bazel defined a [backwards compatibility policy](https://docs.bazel.build/versions/master/backward-compatibility.html)
+(see [https://www.bazel.build/breaking-changes-guide.html](https://www.bazel.build/breaking-changes-guide.html) for how
+to roll out incompatible changes if you are the author of one).
+This page summarized best practices on how to test and migrate your project with upcoming incompatible
+changes and how to provide feedback to the incompatible change authors.
+
+
+## Recommended migration process
+
+Bazel backwards compatibility policy is designed to avoid _upgrade cliffs_:
+any project can be prepared for the next Bazel release without breaking compatibility with the current release.
+
+We recommend the following process for project migration:
+
+
+1. Assume that your project already works with a given Bazel release, say 0.26, and you want to prepare
+   for the next release, say 0.27
+2. Find all incompatible changes for which the migration can be started: they are marked with 
+   "migration-\<release\>" label on GitHub, for example "[migration-0.26](https://github.com/bazelbuild/bazel/issues?utf8=%E2%9C%93&q=label%3Amigration-0.26+)".
```

Comment:
why? the GitHub list is up-to-date, and bazelisk will help with avoiding the copy-pasting

Reply:
The instruction literally says "Find all incompatible changes [...] on GitHub". This is not a nice workflow.
If you want users to run Bazelisk, remove this line because it's misleading.

Reply:
Why? GitHub issue list is an authoritative, up-to-date list of flags. Even if there is a copy-paste-able list somewhere, just copy-pasting the list will not help - users need to test the individual flags, so the workflow you suggest is:
- copy-paste the list into a temporary file
- copy-paste individual flags from that temporary file
- in case of problems, search for information in GitHub tracker (I guess that means "search for an issue on GitHub mentioning that flag"?)

Since you have to come back to GitHub tracker anyway, much easier just to copy-paste individual flags from there directly; but really, people should build a tool that does it for them; that tool is going to work off GitHub issues, not off the announcement blog post.

To be clear, this document is _guidance_, it is not a set of steps we expect people to follow manually; people should figure out how to automate this for their projects (otherwise, it is madness)

Reply:
When it says: "Find all flags [...] For each of them, build your project with that flag enabled", I read this as a list of steps to follow.

Reply:
> When it says: "Find all flags [...] For each of them, build your project with that flag enabled", I read this as a list of steps to follow.

but not necessarily _manually_, and any automated solution should follow the GitHub list.
(and even if you try to do it manually, GitHub issues list is not a bad choice since all information on one page)